sourcecode/npm/components/src/components/AuthWrapper.jsx

```diff
@@ -106,9 +125,40 @@ const AuthWrapper = ({ children }) => {
                 </div>
                 <div style={{marginTop: 40}}>
                     <Button
+                    type="submit"
                         color="primary"
                         variant="contained"
                         onClick={() => {
+                            if (!firstName) {
+                                setfirstNameErrorText("Please enter first name");
+                            }else {
+                                setfirstNameErrorText("");
+                            }
+
+                            if (!lastName) {
+                                setlastNameErrorText("Please enter last name");
+                            }else {
+                                setlastNameErrorText("");
+                            }
+                            
+                            if (!email) {
+                                setemailErrorText("Please enter email");
+                            }else if (email && validateEmail(email)) {
+                                setemailErrorText("Invalid email address");
+                            }else {
+                                setemailErrorText("");
+                            }
+                            
+                            if(validateUserId(id)) {
+                                setidErrorText("Please enter a valid numeric user id");
+                            }else {
+                                setidErrorText("");
+                            }
+                            
+                            if(!firstName || !lastName || !email || validateUserId(id) || validateEmail(email)){
```

Comment:
Instead of repeating all the checks, a single variable that's set to true if an error path is entered would be preferable. For example:

```js
let error = false;

if (!firstName) {
    setFirstNameErrorText("Please enter first name");
    error = true;
} else {
    setFirstNameErrorText("");
}

// ... all the other checks here

if (error) {
    return;
}
```

This way, if any check has to be changed, it only has to be changed in one place.